src/flask/_static_mixin.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import annotations
+
+import os
+import typing as t
+
+from .helpers import send_from_directory
+
+if t.TYPE_CHECKING:  # pragma: no cover
+    from .wrappers import Response
+
+
+class ScaffoldMixin(t.Protocol):
+    has_static_folder: bool
+    root_path: str
+    static_folder: str | None
+
+    def get_send_file_max_age(self, filename: str | None) -> int | None:
+        ...
+
+
+class _StaticMixin:
```

Comment:
Remove, implement in Blueprint and Flask (dupe is ok).
Note "This isn't Sansio, hence dupe in docstring"